Sources/protoc-gen-swift/MessageGenerator.swift

```diff
@@ -718,6 +813,116 @@ class MessageGenerator {
+
+      var functionStarted = false;
+      func ensureFunctionStarted() {
+        if functionStarted { return }
+        functionStarted = true
+        p.print("\npublic var isInitialized: Bool {\n")
+        p.indent()
+        if storage != nil {
+          p.print("return _storage.isInitialized\n")
+        }
+      }
+
+      if isExtensible {
+        ensureFunctionStarted()
+        if storage == nil {
+          p.print("if !extensionFieldValues.isInitialized {return false}\n")
+        }
+      }
+
+      if !isProto3 {
+        // Only proto2 syntax can have field presence (required fields); ensure required
+        // fields have values.
+        for f in fields {
+          if f.descriptor.label != .required {
+            continue
+          }
+          ensureFunctionStarted()
+          if storage != nil {
+            // No need to check other fields, using storage deferred to the
+            // storage_.isInitialized.
+            break
+          } else {
+            p.print("if \(f.swiftStorageName) == nil {return false}\n")
+          }
+        }
+      }
+
+      // Check that all non-oneof embedded messages are initialized.
+      for f in fields {
+        if f.fieldHoldsMessage && f.oneof == nil && HasRequiredFields(f.descriptor, context: context) {
+          ensureFunctionStarted()
+          if storage != nil {
+            // No need to check other fields, using storage deferred to the
+            // storage_.isInitialized.
+            break
+          } else {
+            if f.isRepeated {
+              p.print("if !SwiftProtobuf.Internal.areAllInitialized(\(f.swiftName)) {return false}\n")
+            } else {
+              p.print("if let v = \(f.swiftName), !v.isInitialized {return false}\n")
+            }
+          }
+        }
+      }
+
+      // Check the oneofs using a switch so we can be more efficent.
+      for oneofField in oneofs {
+        var hasRequiredFields = false
+        for f in oneofField.fields {
+          if f.descriptor.isMessage && HasRequiredFields(f.descriptor, context: context) {
+            hasRequiredFields = true
+            break
+          }
+        }
+        if !hasRequiredFields {
+          continue
+        }
+
+        ensureFunctionStarted()
+        if storage != nil {
+          // No need to check other fields, using storage deferred to the
+          // storage_.isInitialized.
+          break;
+        }
+
+        p.print("switch \(oneofField.descriptor.swiftFieldName) {\n")
+        var needsDefault = false
+        for f in oneofField.fields {
+          if f.descriptor.isMessage && HasRequiredFields(f.descriptor, context: context) {
+            p.print("case .\(f.swiftName)(let v):\n")
+            p.indent()
+            p.print("if !v.isInitialized {return false}\n")
+            p.outdent()
+          } else {
+            needsDefault = true
+          }
+        }
+        p.print("case .None:\n")
```

Comment:
When #133 is done, we can drop the `.None` cases from there, shortening them a hair.